include/pybind11/pybind11.h

```diff
@@ -1406,6 +1406,9 @@ class class_ : public detail::generic_type {
             );
         }
         v_h.value_ptr() = nullptr;
+        if (PyErr_Occurred()) {
+            PyErr_WriteUnraisable(v_h.type ? (PyObject*)v_h.type->type : nullptr);
```

Comment:
What's the reason for passing the `type` rather than the value here?

Reply:
`v_h.type->type` is the class, not the exception.

I assumed that if I hit an error here, then the value no longer existed so a `__repr__` wouldn't be sensible? I'm not sure if the name is still around, though I guess it must be?

Though, you're right, it would be good to do something more sensible. Python does this:

```
$ cat t.py
class X:
    def __del__(self):
        raise ValueError("something")
X()
$ python t.py
Exception ignored in: <function X.__del__ at 0x7fd070258f28>
Traceback (most recent call last):
  File "t.py", line 3, in __del__
    raise ValueError("something")
ValueError: something
```
As mentioned above, here's what I currently have:

```
Exception ignored in: <class 'pybind11_tests.class_.PyRaiseDestructor'>
Traceback (most recent call last):
  File "<snip>/pybind11/tests/test_class.py", line 389, in test_pyexception_destructor
    m.PyRaiseDestructor()
ValueError: unraisable error
```

I suppose we could add a name and a pointer value?

Reply:
Actually it's probably good as is, since printing the name of the offending class gives a good idea as to where the occurred. 

We definitely don't want to call any extra functions in this type of code that code error out.